sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -32,7 +45,23 @@ export class EducationProgramOfferingController {
     @Body() payload: SaveEducationProgramOfferingDto,
     @Param("locationId") locationId: number,
     @Param("programId") programId: number,
+    @UserToken() userToken: IInstitutionUserToken,
   ): Promise<number> {
+    const requestedLoc = await this.locationService.getInstitutionLocationById(
+      locationId,
+    );
+    if (
+      userToken.authorizations.institutionId !== requestedLoc.institution.id
+    ) {
+      throw new ForbiddenException();
+    }
+    const requestProgram = await this.programService.getInstitutionProgram(
+      programId,
+      userToken.authorizations.institutionId,
+    );
+    if (!requestProgram) {
+      throw new ForbiddenException();
```

Comment:
At this point, the institution access was already validated, I would recommend a throw a NotFoundException.